lib/logstash/codecs/cef.rb

```diff
@@ -20,31 +15,45 @@ def initialize(params={})
 
   public
   def decode(data)
-    # Need to break out the headers, then return the headers as individual fields, and the extension to be processed by a filter (ie: KV)
-    # %{SYSLOGDATE} %{HOST} CEF:Version|Device Vendor|Device Product|Device Version|SignatureID|Name|Severity|Extension
-    event = LogStash::Event.new()
-    if @syslog
-      @logger.debug("Expecting SYSLOG headers")
-      event['syslog'], data = data.split('CEF:', 2)
-      # Since we have the syslog headers, lets pull them out first and put them into their own field to be handled
-    else 
-      # We don't have syslog headers, so we just need to remove CEF:
-      data.sub! /^CEF:/, ''
-    end #if @syslog
-    # Now, break out the rest of the headers
-    event['cef_version'], event['cef_vendor'], event['cef_product'], event['cef_device_version'], event['cef_sigid'], event['cef_name'], event['cef_severity'], event['message'] =  data.scan /(?:[^\|\\]|\\.)+/
-    # Now, try to break out the Extension Dictionary
-    message=event['message']
-    if message.to_s.strip.length != 0
+    # Strip any quotations at the start and end, flex connectors seem to send this
+    if data[0] == "\""
+      data = data[1..-2]
+    end
+    event = LogStash::Event.new
+
+    # Split by the pipes
+    event['cef_version'], event['cef_vendor'], event['cef_product'], event['cef_device_version'], event['cef_sigid'], event['cef_name'], event['cef_severity'], message = data.split /(?<!\\)[\|]/
+
+    # Try and parse out the syslog header if there is one
+    if event['cef_version'].include? ' '
+      event['syslog'], unused, event['cef_version'] = event['cef_version'].rpartition(' ')
+    end
+
+    # Get rid of the CEF bit in the version
+    event['cef_version'].sub! /^CEF:/, ''
+
+    # Strip any whitespace from the message
+    message = message.to_s.strip
+
+    # If the last KVP has no value, add an empty string, this prevents hash errors below
+    if message.end_with?("=")
+      message=message + ' '
+    end
+
+    # Now parse the key value pairs into it
+    extensions = {}
+    if message.length != 0 and message.include? "="
       message = message.split(/ ([\w\.]+)=/)
+      key, value = message.shift.split('=', 2)
+      extensions[key] = value
 
-      key, value = message.shift.split('=',2)
-      @logger.debug(message)
-      kv = Hash[*message]
-      @logger.debug(kv)
-      addKey(kv,key,value)
-      event.to_hash.merge!(Hash[kv.map{ |k,v| ["cef_ext_"+k,v] }])
-    end #
+      Hash[*message].each{|k, v| 
```

Comment:
just for style consistency, can you either switch to 1 line {} or block do;end such as?

```ruby
Hash[*message].each{|k, v| extensions[k] = v }
```

or 

```ruby
Hash[*message].each do |k, v|
  extensions[k] = v
end
```